### PR TITLE
added shape, shape offsets, and headings.

### PR DESCRIPTION
**Pull request by gknisely**
@dgearhart @dnesbitt61 @kevinkreiser

### CODE REVIEW

File: src/thor/trippathbuilder.cc [hunk 1]
```diff
@@ -74,31 +81,42 @@ TripPath TripPathBuilder::Build(GraphReader& graphreader,
...
-    names = graphtile->GetNames(directededge->edgedataoffset(), names);
-    for (auto& name : names) {
-      std::cout << "   name=" << name << std::endl;
-    }
-    const std::shared_ptr<EdgeInfo> edgeinfo = graphtile->edgeinfo(
-        directededge->edgedataoffset());
-    // TODO - rm later
-    //edgeinfo->ToOstream();
-*/
+    trip_edge->set_begin_heading(PointLL::HeadingAtEndOfPolyline(points,30));
```
Comment (dgearhart) [inline src/thor/trippathbuilder.cc]: looks like copy/paste error - should be setting the end heading correct?

File: src/thor/trippathbuilder.cc [hunk 2]
```diff
@@ -74,31 +81,42 @@ TripPath TripPathBuilder::Build(GraphReader& graphreader,
...
+      //Add shape in the reverse direction.
+      std::reverse(points.begin(), points.end());
+
+    }
+
+    trip_edge->set_begin_shape_index(trip_shape.size());
+
+    unsigned int n = 0;
 
+    for (const auto& pt : points) {
```
Comment (kevinkreiser) [inline src/thor/trippathbuilder.cc]: why use a foreach here when you need the index so bad? if there are dups we shoudl remove them at data creation time.

shouldnt we just do:
   if(is_reverse)
      trip_shape.assign(edgeinfo->shape().rbegin(), edgeinfo->shape().rend())
   else
      trip_shape.assign(edgeinfo->shape().begin(), edgeinfo->shape().end())

you are currently doing two copies here

**Comment by kevinkreiser** [comment]
other than that shape thing :shipit:

**Comment by dgearhart** [comment]
shape fix and then :+1:

**Comment by dgearhart** [comment]
ship

**Pull request merged**